### PR TITLE
more strict 3 + deprecate connectDEBUG

### PR DESCRIPTION
**Pull request by chrisnojima**
More strict work, plus a better debug flow for connect
if there are mismatches `connect` will emit a type like this:
Screen Shot 2020-01-07 at 4 17 09 PM

which means `message` should be `MessageText` but merge returned `number`
`TS correct` means it matches

### CODE REVIEW

File: shared/profile/showcase-team-offer/container.tsx [hunk 1]
```diff
@@ -11,7 +11,7 @@ type OwnProps = {}
 
 export default Container.connect(
   (state: Container.TypedState) => ({
-    _waiting: state.waiting,
+    _waiting: state.waiting.counts,
```
Comment (chrisnojima) [inline shared/profile/showcase-team-offer/container.tsx]: actual bug!

File: shared/settings/account/email-phone-row.tsx [hunk 2]
```diff
@@ -303,6 +300,7 @@ const ConnectedEmailPhoneRow = Container.namedConnect(
         onVerify: dispatchProps.email.onVerify,
         primary: stateProps._emailRow.isPrimary,
         searchable,
+        superseded: false,
```
Comment (chrisnojima) [inline shared/settings/account/email-phone-row.tsx]: actually missing!

File: shared/override-d.ts/react-redux/index.d.ts [hunk 3]
```diff
@@ -227,32 +227,13 @@ export type ResolveArrayThunks<TDispatchProps extends ReadonlyArray<any>> = TDis
...
-  <TOwnProps, TStateProps, TDispatchProps, TMergedProps>(
-    mapStateToProps: MapStateToProps<TStateProps, TOwnProps>,
-    mapDispatchToProps: MapDispatchToProps<TDispatchProps, TOwnProps>,
-    mergeProps: MergeProps<TStateProps, TDispatchProps, TOwnProps, TMergedProps>,
-    options?: Options<TypedState, TStateProps, TOwnProps, TMergedProps>
-  ): ConnectedComponentTypeDEBUG<TMergedProps, TOwnProps>
-}
-
-export const connectDEBUG: ConnectDEBUG
+  : {
```
Comment (jakob223) [inline shared/override-d.ts/react-redux/index.d.ts]: this won't show missing props, right? Maybe we could make it `{ missing: Exclude<etc>, present: WhatYouHaveNow }`?

File: shared/util/container.tsx [hunk 4]
```diff
@@ -81,7 +81,7 @@ export const timeoutPromise = (timeMs: number) =>
     setTimeout(() => resolve(), timeMs)
   })
 
-export {default as connect, namedConnect, connectDEBUG} from './typed-connect'
+export {default as connect, namedConnect} from './typed-connect'
```
Comment (jakob223) [inline shared/util/container.tsx]: 🎉

File: shared/override-d.ts/react-redux/index.d.ts (hunk 3 above)
Comment (chrisnojima) [reply shared/override-d.ts/react-redux/index.d.ts]: good catch, i just include this info in the latest commit changes:

Screen Shot 2020-01-08 at 9 46 33 AM

**Pull request merged**